From: Ralf Baechle <ralf@linux-mips.org>
Date: Wed, 3 Dec 2008 16:33:17 +0000 (+0000)
Subject: MIPS: o32: Fix number of arguments to splice(2).
X-Git-Url: http://www.linux-mips.org/git?p=linux.git;a=commitdiff_plain;h=532f50e0e56bab50d88d3056e2d69e8db5b1feea

MIPS: o32: Fix number of arguments to splice(2).

The syscall code was assuming splice only takes 4 arguments so no stack
arguments were being copied from the userspace stack to the kernel stack.
As the result splice was likely to fail with EINVAL.

Signed-off-by: Ralf Baechle <ralf@linux-mips.org>
(cherry picked from commit 83a5a562925a5b606f9922b0c6888b64d067ceb0)
---

diff --git a/arch/mips/kernel/scall32-o32.S b/arch/mips/kernel/scall32-o32.S
index fb116bf..9cc3e54 100644
--- a/arch/mips/kernel/scall32-o32.S
+++ b/arch/mips/kernel/scall32-o32.S
@@ -639,7 +639,7 @@ einval:	li	v0, -ENOSYS
 	sys	sys_pselect6		6
 	sys	sys_ppoll		5
 	sys	sys_unshare		1
-	sys	sys_splice		4
+	sys	sys_splice		6
 	sys	sys_sync_file_range	7	/* 4305 */
 	sys	sys_tee			4
 	sys	sys_vmsplice		4
